### ISSUE TITLE
OSM login fails with stock browser in Sailfish OS

### ISSUE THREAD

**Issue by nospam3059** [westnordost/StreetComplete#1760]
**How to Reproduce**

0. Install StreetComplete on a Sailfish OS device and launch it.
1. Select the option to OAuth with OSM.
2. Login in the browser window opened.
3. Leave all three checkboxes checked and click "Grant Access"
4. See the "Grant Access" button disappear.
5. Wait indefinitely. (The browser does not even attempt to report back)

I've tried switching the browser from private (incognito) to normal mode.

Installing and using firefox made step 5 above succeed. Still reporting as an issue, for two reasons.

Firstly, issue #1597 mentions Vespucci working differently. As can be seen from the first of my screenshots, I've successfully authorized that app without installing any other browser. So it should be possible in StreetComplete too, yet maybe not worth the effort.

Secondly, documenting my experience like this might possibly help anyone else on Sailfish to find the workaround. It took me a while to even realize it could be done that way.

Since authorizing is done only once, the user count running Sailfish OS can be assumed to be small and a workaround exist, I guess the issue could be closed immediately, unless someone feels a great desire to dive into it.

**Versions affected**

Sailfish OS Nuuksio (3.2.1.20)
StreetComplete v18.0

![Skärmbild_20200309_002](https://user-images.githubusercontent.com/61987353/76247205-75616400-623f-11ea-8b2e-8fd50e534d5f.png)
![Skärmbild_20200309_003](https://user-images.githubusercontent.com/61987353/76247250-8316e980-623f-11ea-9c2b-acdc4e7d1212.png)
![Skärmbild_20200309_004](https://user-images.githubusercontent.com/61987353/76247259-87430700-623f-11ea-88bf-a92b03a78b0f.png)

**Comment by westnordost** [comment]
Duplicate of #413. Will be changed/fixed for the next version
